lib/src/svg/theme.dart

```diff
@@ -0,0 +1,30 @@
+import 'dart:ui';
+
+/// A theme used when decoding an SVG picture.
+class SvgTheme {
+  /// Instantiates an SVG theme with the [colorFilter] and [currentColor].
+  const SvgTheme({
+    this.colorFilter,
+    this.currentColor,
```

Comment:
It looks like this is present here and as an individual property on the widget. 

I think I would have expected this to be an inherited widget, or perhaps for there to be a `DefaultSvgTheme` widget that's an inherited widget that the child widget(s) can use if one isn't specified.

Reply:
I have created a `DefaultSvgTheme` - an inherited widget that provides `SvgTheme` to descendant widgets. 

I am now thinking of a clean way to access the provided theme in `SvgPicture`. The theme needs to be passed to `svg.svgPictureDecoder` (so it can be provided to `SvgParser`) and at the moment all decoders are initialized in `SvgPicture` constructors. To access the theme from context, we could probably move the initialization of decoders to `initState`. Do you think that would be okay?

Reply:
Yes - all of this plays into the caching concerns I was mentioning. Today we assume that the picture is static and only the width/height will change -but now properties related to decoding can change.